### PR TITLE
Adjust IsMetadataVirtual assertion in MethodToClassRewriter

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/73563

`IsMetadataVirtual` previously could either:
- give a final answer but this required it to be called **after** `ForceComplete` (which may initialize it in `SourceMemberContainerTypeSymbolSynthesizeInterfaceMemberImplementation`)
- give a partial answer (by passing `ignoreInterfaceImplementationChanges`)

Previously, in a scenario where `comp2` references `comp1` as a compilation reference and we emit `comp2` and then emit `comp1`:
1. emitting `comp2` called `MethodToClassRewriter.VisitCall` which called `IsMetadataVirtual` on a not-yet-completed symbol from `comp1`. The read from `IsMetadataVirtual` set the "locked" flag.
2. emitting `comp1` completed that symbol from `comp1` which would call `EnsureMetadataVirtual`. This asserted since the "locked" flag was already set

Now, the assertion in `MethodToClassRewriter.VisitCall` eagerly forces the completion of the symbol from `comp1`.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Emitter/Model/MethodSymbolAdapter.cs [hunk 1]
```diff
@@ -658,12 +658,24 @@ internal virtual bool IsMetadataFinal
         /// signature).
         /// WARN WARN WARN: We won't have a final value for this until declaration
         /// diagnostics have been computed for all <see cref="SourceMemberContainerTypeSymbol"/>s, so pass
-        /// ignoringInterfaceImplementationChanges: true if you need a value sooner
+        /// option: <see cref="IsMetadataVirtualOption.IgnoreInterfaceImplementationChanges"/> true if you need a value sooner
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Emitter/Model/MethodSymbolAdapter.cs:661]: >true [](http://example.com/codeflow?start=95&length=4)

Delete? This looks like an artifact from the old state of the comment.

**Comment by AlekseyTs** [comment]
It looks like a couple of tests in `RefStructInterfacesTests.cs` were disabled because of this issue. I think we should enable them in this PR.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review. Thanks

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review. Thanks

**Review by 333fred — APPROVED** [review]

**Pull request merged**